### PR TITLE
docs: refactor contributing guidelines - add project structure

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: docs/dev/project-structure.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+- `05-port`: This package is an implementation for Cosmos SDK-based chains of [ICS-05](https://github.com/cosmos/ibc/tree/main/spec/core/ics-005-port-allocation). This implements the port allocation system by which modules can bind to uniquely named ports.
+- `23-commitment`: This package is an implementation for Cosmos SDK-based chains of [ICS-23](https://github.com/cosmos/ibc/tree/main/spec/core/ics-023-vector-commitments). This implementation defines the functions required to prove inclusion or non-inclusion of particular values at particular paths in state.
+- `24-host`: This package is an implementation for Cosmos SDK-based chains of [ICS-24](https://github.com/cosmos/ibc/tree/main/spec/core/ics-024-host-requirements).
+
+### `apps`
+
+- `transfer`: This is the Cosmos SDK implementation of the [ICS-20](https://github.com/cosmos/ibc/tree/main/spec/app/ics-020-fungible-token-transfer) protocol, which enables cross-chain fungible token transfers. For more information, read the [module's docs](../apps/transfer/overview.md)
```
Comment (chatton) [inline docs/dev/project-structure.md:22]: I think we should rename this one to `20-transfer`, it seems to be the odd one out at the moment! (not in this PR of course!)

**Review by chatton — APPROVED** [review]
LGTM!

**Review by colin-axner — APPROVED** [review]
Wonderful!

**Review by damiannolan — APPROVED** [review]
Nice!

File: docs/dev/project-structure.md (hunk 1 above)
Comment (crodriguezvega) [reply docs/dev/project-structure.md:22]: I agree. Something to do at some point!

**Pull request merged**